pkg/k8s/client.go

```diff
@@ -344,7 +345,7 @@ func (c client) UpdateStatus(resource interface{}) (metav1.Object, error) {
 
 // ServiceToMeshServices translates a k8s service with one or more ports to one or more
 // MeshService objects per port.
-func ServiceToMeshServices(c Controller, svc corev1.Service) []service.MeshService {
+func ServiceToMeshServices(svc corev1.Service, endpointsGetter func(service.MeshService) (*corev1.Endpoints, error)) []service.MeshService {
```

Comment:
A getter feels like an unnecessary layer of indirection. We're already operating on a k8s corev1.Service, so this should know how to translate between headless and non-headless. 

It likely makes sense to add this as a method on client instead

Reply:
The reason the getter was added was because the verifier package needs to call this method without initializing an entire controller client. See [this comment](https://github.com/openservicemesh/osm/pull/4704#discussion_r869666034) for context

Reply:
I'd like to push back against that more still. Can we refactor so that it does have a client?

Regarding that comment, and as a general rule, I think we should avoid pushing changes we think are currently code smells with the excuse that we will refactor soon. Too often things get left untouched, and if we can refactor soon, I think we can refactor now.

if it makes the PR too large, maybe we can get a refactoring PR in now, and then merge/rebase the changes here

Reply:
I agree with you in general. The complexity with this particular subject is that the verifier shouldn't need a whole k8s controller with informers and such. At this point, I truly wonder if de-duplication is worth it; yes the logic is the same, but without using indirection such as this getter function, it's impossible to generalize due to the various locations data is coming from (i.e. controller vs raw client-go client)

Reply:
Spoke offline with Sean, decided to circle back to this later. For now, I'm going to err on the side of the [WET principle](https://betterprogramming.pub/when-dry-doesnt-work-go-wet-6befda0444bf) and duplicate these for loops